### DISCUSSION TITLE
How to submit ENTER keypress?

### DISCUSSION THREAD

**Discussion by Revadike** [leancodepl/patrol#1516] [Q&A]
I have a webview with a submit button, that somehow cannot be found by patrol. So, as a workaround, I thought perhaps I can just submit the ENTER keypress, but I can't make that work either... I would love some assistance.

**Comment by Revadike** [comment]
some things I tried...

```
      // await $.native.tap(Selector(text: 'Inloggen'));
      // await $.native.tap(Selector(textContains: 'Inloggen'));
      // await $('Inloggen').tap();
      // await $.native.enterTextByIndex('\n', index: 1);
      await $.tester.testTextInput.receiveAction(TextInputAction.done);
```

nothing seems to work

**Comment by bartekpacia** [comment]
Sounds like you'd want to have [UiAutomator#pressEnter()](https://developer.android.com/reference/androidx/test/uiautomator/UiDevice#pressEnter()) exposed from Dart, don't you?

I don't know if it has any equivalent on iOS.

**Reply by Revadike** [reply]
How can I do this with patrol? Excuse my lack of knowledge

**Reply by bartekpacia** [reply]
Right now you cannot, but we could expose that functionality.

**Reply by Revadike** [reply]
Yeah, that seems like a useful feature.

**Reply by Revadike** [reply]
Meanwhile, any idea why Patrol cannot find the login button to tap from https://inside.heezenbedrijven.nl/inloggen/ ?

**Reply by martingeorgiu** [reply]
Yeah that would be super helpful for us. As mentioned in this issue https://github.com/leancodepl/patrol/issues/7 we cannot use any other way using standard Patrol `Selector`

**Comment by pandoras-toolbox** [comment]
I have done something like this, works on Android in the webview, not tested on iPhone yet:
```
await $.native.enterText(Selector(text: 'Username'), text: 'admin');
await $.native.enterText(Selector(text: 'Password'), text: 'secret');
await $.native.tap(Selector(className: 'android.widget.Button', instance: 2));
```
It is not nice having to use instance=2, but at least it works.

**Comment by andy-todd-dev** [comment]
Any update on this? Is it now possible?
